src/pkg/build/assemblers/assembler_sif.go

```diff
@@ -129,3 +136,41 @@ func (a *SIFAssembler) Assemble(b *types.Bundle, path string) (err error) {
 
 	return
 }
+
+// changeOwner check the command being called with sudo with the environment
+// variable SUDO_COMMAND. Pattern match that for the singularity bin
+func changeOwner() (int, int, bool) {
+	var uid, gid int
+	var chown bool
+	var err error
+
+	r := regexp.MustCompile("(singularity)")
+	sudoCmd := os.Getenv("SUDO_COMMAND")
+	if r.MatchString(sudoCmd) {
+		if syscall.Getuid() == 0 && os.Getenv("SUDO_USER") != "root" {
+			_uid := os.Getenv("SUDO_UID")
+			_gid := os.Getenv("SUDO_GID")
+			if _uid == "" || _gid == "" {
+				sylog.Errorf("Env vars SUDO_UID or SUDO_GID are not set, won't call chown over built SIF")
+
+				return 0, 0, false
+			}
+
+			uid, err = strconv.Atoi(_uid)
+			if err != nil {
+				sylog.Errorf("Error while calling strconv: %v", err)
+
+				return 0, 0, false
+			}
+			gid, err = strconv.Atoi(_gid)
+			if err != nil {
+				sylog.Errorf("Error while calling strconv : %v", err)
```

Comment:
I would say `Warningf` since these are not "errors" which will cause singularity to exit

Reply:
Same as the other `Errorf` spots as well